heartbeat/monitors/wrappers/monitorstate/monitorstate.go

```diff
@@ -44,7 +44,7 @@ func newMonitorState(sf stdfields.StdMonitorFields, status StateStatus, ctr int,
 		// ID is unique and sortable by time for easier aggregations
 		// Note that we add an incrementing counter to help with the fact that
 		// millisecond res isn't quite enough for uniqueness (esp. in tests)
-		ID:              fmt.Sprintf("%s-%s-%x-%x", sf.ID, sf.Location, now.UnixMilli(), ctr),
+		ID:              fmt.Sprintf("%s-%s-%x-%x", sf.ID, sf.RunFrom, now.UnixMilli(), ctr),
```

Comment:
Need to account for missing `RunFrom` config. This is what I see in states

```
  "state": {
    "started_at": "2022-09-13T11:22:16.489384-07:00",
    "up": 0,
    "ends": null,
    "id": "my-monitor-%!s(*config.LocationWithID=<nil>)-183381669a9-0",
    "duration_ms": 20003,
    "status": "down",
    "checks": 3,
    "down": 3,
    "flap_history": []
  },
```

We can default to `unknown` location ?

Reply:
Also it should be RunFrom.id, since its using interface now.